x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -502,16 +502,12 @@ export class VectorLayer extends AbstractLayer {
   }
 
   async getPropertiesForTooltip(properties) {
-    const tooltipsFromSource =  await this._source.filterAndFormatProperties(properties);
-
-    //add tooltips from joins
-    return this._joins.reduce((acc, join) => {
-      const propsFromJoin = join.filterAndFormatPropertiesForTooltip(properties);
-      return {
-        ...propsFromJoin,
-        ...acc,
-      };
-    }, { ...tooltipsFromSource });
+    const tooltipsFromSource =  await this._source.filterAndFormatPropertiesToHtml(properties);
+    for (let i = 0; i < this._joins.length; i++) {
```

Comment:
Why did you switch from reduce to old fashioned for loop? Seems like our coding standards state preference for declarative constructs like `forEach` and `reduce` versus imperative constructs like `for`. Can we keep using `reduce` here?

Reply:
makes the use of `await` more simple, as it applies to the scope in the current block. Array#reduce does not await return of the accumulating function.